allennlp/training/trainer.py

```diff
@@ -116,11 +124,28 @@ def train(self) -> None:
                     val_output_dict = self._model.forward(**tensor_batch)
                     loss = val_output_dict["loss"]
                     val_loss += loss.data.cpu().numpy()
-
-            # TODO(Mark): Add user specified metrics here, maybe a "metrics" key?
-            logger.info("Training Loss: %3f    Validation Loss: %3f ", train_loss, val_loss)
-            if self._serialization_prefix:
-                self._save_checkpoint(epoch)
+                val_metrics = self._model.get_metrics(reset=True) or {"loss": val_loss}
+
+                message_template = "Training %s : %3f    Validation %s : %3f "
+                for name, value in metrics.items():
+                    logger.info(message_template, name, value, name, val_metrics[name])
+                logger.info(message_template, "Loss", train_loss, "Loss", val_loss)
```

Comment:
The idea with putting the loss in the metrics dictionary is that you can now just remove this line.